### ISSUE TITLE
Swipe gesture to pop?

### ISSUE THREAD

**Issue by aksonov** [ericvicenti/navigation-rfc#43]
I doesn't work for latest 0.22.0-rc, it allows to swipe, but after touch finishes returns to previous state (no pop).

**Comment by aksonov** [comment]
![simulator screen shot 09 mar 2016 17 31 35](https://cloud.githubusercontent.com/assets/1321329/13642523/24e8170a-e61d-11e5-909b-7b726ec2bdcc.png)

**Comment by hedgerwang** [comment]
This example `NavigationAnimatedExample` does not handle the action `{type: 'back'}` that is invoked by the `NavigationCard`.

We could change the reducer to handle such action.

```
function createReducer(initialState) {
  return (currentState, action) => {
    switch (action.type) {
      case 'push':
        return NavigationStateUtils.push(currentState, {key: action.key});

      case 'back':
        return currentState.index > 0 ?
          NavigationStateUtils.pop(currentState) :
          currentState;

      default:
        return currentState;
    }
  };
}

const NavigationBasicReducer = createReducer({
  key: 'AnimatedExampleStackKey',
  index: 0,
  children: [
    {key: 'First Route'},
  ],
});

```

**Comment by hedgerwang** [comment]
Or the `NavigationCard` could just invoke the action `{type: 'BackAction'}`.

**Comment by aksonov** [comment]
Thanks, i've tried your reducer but got blank screen. Current example reducer is following :

```
const NavigationBasicReducer = NavigationReducer.StackReducer({
    getPushedReducerForAction: (action) => {
        if (action.type === 'push') {
            return (state) => state || {key: action.key};
        }
        return null;
    },
    getReducerForState: (initialState) => (state) => state || initialState,
    initialState: {
        key: 'AnimatedExampleStackKey',
        index: 1,
        children: [
            {key: 'First Route'},
            {key: 'Second Route'},
        ],
    },
});

```

How to add handler for 'back'? Strange, state is null within getPushedReducerForAction.

How to invoke {type:'BackAction'} for NavigationCard?

**Comment by aksonov** [comment]
I've solved the problem by duplicating action for 'BackAction' within NavigationStackReducer:

```
      case 'back':
        if (lastParentState.index === 0 || lastParentState.children.length === 1) {
          return lastParentState;
        }
        return NavigationStateUtils.pop(lastParentState);
      case 'BackAction':
        if (lastParentState.index === 0 || lastParentState.children.length === 1) {
          return lastParentState;
        }
        return NavigationStateUtils.pop(lastParentState);

```

Is there any reason why swipe generates 'back' but not 'BackAction'?

**Comment by hedgerwang** [comment]
We haven't formalized the action for the backward gesture. It should be `BackAction`.

I'm not familiar with `NavigationReducer.StackReducer`. Somehow it does not have a reducer for `pop` action.

cc @ericvicenti

**Comment by ericvicenti** [comment]
Fixed here:

https://github.com/facebook/react-native/commit/b1090bfae145e6fa7a18a851fd78f966113a07c4

Sorry we couldn't get this fixed in time for 0.22!

@hedgerwang, we really need to be careful with breaking changes!
